tests/Integration/Behaviour/Features/Scenario/Product/Shop/shop_management.feature

```diff
@@ -616,3 +617,25 @@ Feature: Copy product from shop to shop.
       | product1LBlack | Size - L, Color - Black |           | [Size:L,Color:Black] | 0               | 0        | false      |
       | product1LBlue  | Size - L, Color - Blue  |           | [Size:L,Color:Blue]  | 0               | 0        | false      |
     And product "product1" should have no combinations for shops "shop2"
+
+  Scenario: Product images are copied/deleted when product is being copied/deleted to/from shop.
+    Given I add product "product1" with following information:
+      | name[en-US] | universal T-shirt |
+      | type        | standard          |
+    And product product1 type should be standard
+    And product "product1" should have no images for shops "shop1"
+    And product "product1" is not associated to shop "shop2"
+    When I add new image "image1" named "app_icon.png" to product "product1" for shop "shop1"
+    Then product "product1" should have following images for shops "shop1":
+      | image reference | is cover | legend[en-US] | legend[fr-FR] | position | image url                            | thumbnail url                                      |
+      | image1          | true     |               |               | 1        | http://myshop.com/img/p/{image1}.jpg | http://myshop.com/img/p/{image1}-small_default.jpg |
+    When I copy product "product1" from shop "shop1" to shop "shop2"
+    Then product "product1" should have following images for shops "shop1,shop2":
+      | image reference | is cover | legend[en-US] | legend[fr-FR] | position | image url                            | thumbnail url                                      |
+      | image1          | true     |               |               | 1        | http://myshop.com/img/p/{image1}.jpg | http://myshop.com/img/p/{image1}-small_default.jpg |
+    When I delete product "product1" from shops "shop2"
+    Then product "product1" should have no images for shops "shop2"
+    And product "product1" is not associated to shop "shop2"
+    And product "product1" should have following images for shops "shop1":
```

Comment:
You need to copy it again that's when the bug was present in our case (some rows left in the tables cause unicity exceptions)